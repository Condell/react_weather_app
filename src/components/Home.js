import React from "react";
import styled from "styled-components";

import pattern from "../images/pattern.svg";
import Input from "./shared/Input";
import Button from "./shared/Button";

const HomeWrapper = styled.div`
  background-image: url(${pattern});
  background-size: 100% 100%;
  background-color: #fe9c42;
  height: 80vh;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  margin-top: 0px;
`;

const Title = styled.h1`
  font-family: 'Abel', sans-serif;
  color: white;
`;

const Home = () =>
  <HomeWrapper>
    <Title>Enter a City and State</Title>
    <Input type="text" placeholder="St. George, Utah" />
    <Button>Get Weather</Button>
  </HomeWrapper>;

export default Home;
