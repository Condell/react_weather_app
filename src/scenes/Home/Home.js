import React from "react";
import styled from "styled-components";

import pattern from "./images/pattern.svg";
import WeatherContainer from "../../components/WeatherFormContainer/WeatherFormContainer";
import Title from "../../components/Title/Title";

const HomeWrapper = styled.div`
  background-image: url(${pattern});
  background-size: 100% 100%;
  background-color: #fe9c42;
  min-height: 80vh;
  display: flex;
  flex-flow: column;
  justify-content: center;
  align-content: center;
  align-items: center;
  text-align: center;
`;

const Home = () =>
  <HomeWrapper>
    <Title whiteFont>Enter a City and State, or a ZIP Code</Title>
    <WeatherContainer />
  </HomeWrapper>;

export default Home;
