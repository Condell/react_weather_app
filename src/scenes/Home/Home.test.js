import React from "react";
import { shallow } from "enzyme";
import 'jest-styled-components'
import toJson from "enzyme-to-json"

import Home from "./Home";

jest.mock("./Home.js", () => "Home")

describe("Home", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Home />);
  });

  it("should render", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});