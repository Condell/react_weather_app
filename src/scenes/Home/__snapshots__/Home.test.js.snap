// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should render 1`] = `
<styled.div>
  <styled.h1
    whiteFont={true}
  >
    Enter a City and State, or a ZIP Code
  </styled.h1>
  <WeatherFormContainer />
</styled.div>
`;
